packages/xsnap/src/avaXS.js

```diff
@@ -21,7 +22,7 @@ const avaHandler = `./avaHandler.cjs`;
 
 /** @type { (ref: string, readFile: typeof import('fs').promises.readFile ) => Promise<string> } */
 const asset = (ref, readFile) =>
-  readFile(new URL(ref, import.meta.url).pathname, 'utf8');
+  readFile(fileURLToPath(new URL(ref, import.meta.url)), 'utf8');
```

Comment:
`asset()` is about reading stuff out of the package. I wonder about using the `node:module` API here.

After thinking it over, I suppose it's more of a change than is necessary to fix this.

But I'll leave my thinking-out-loud here for comparison...


```suggestion
  readFile(nodeRequire.resolve(ref), 'utf8');
```

presuming:

```js
import { createRequire } from 'node:module';

const nodeRequire = createRequire(import.meta.url);
```

Reply:
It occurred to me, and would be useful especially for reaching for assets in other packages.

Reply:
I like it!

Not a blocker, of course. I suppose it'll come up again when we fully expunge `.pathname`